pkg/kubectl/resource/visitor.go

```diff
@@ -386,8 +414,8 @@ type StreamVisitor struct {
 // empty stream is treated as an error for now.
 // TODO: convert ignoreErrors into a func(data, error, count) bool that consumers can use to decide
 // what to do with ignored errors.
-func NewStreamVisitor(r io.Reader, mapper *Mapper, source string, ignoreErrors bool) Visitor {
-	return &StreamVisitor{r, mapper, source, ignoreErrors}
+func NewStreamVisitor(r io.Reader, mapper *Mapper, schema validation.Schema, source string, ignoreErrors bool) Visitor {
```

Comment:
It's unfortunate having to decorate all the visitors like this.  There is no other way today unless we instead made this a generic interface (that allows you to act on the undecoded data prior to operating on it).